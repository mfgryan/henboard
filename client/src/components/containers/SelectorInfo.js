// redux dep
import { connect } from "react-redux";
import sprints from "../../models/sprints.js";
import { closeInfo } from "../../actions/items";
import { moveItem } from "../../actions/items.js";

// component dep
import Selector from "../Selector";

const mapStateToProps = (state, ownProps) => {
    return {
        title: "Send To",
        items: sprints.getSprintArray(state, ownProps.project),
        itemKey: "week"
    };
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        itemClicked: key => {
            dispatch(moveItem({
                project: ownProps.project, 
                name: ownProps.title,
                week: key,
                column: "Todo"
            }));  
            dispatch(
                closeInfo({
                    project: ownProps.project, 
                    name: ownProps.title
                }) 
            );
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Selector);
