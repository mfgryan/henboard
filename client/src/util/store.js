// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js";
import data from "../util/data.js";
import rootReducer from "../reducers/index.js";

const store = createStore(rootReducer,applyMiddleware(logger,validate));

export { store }
