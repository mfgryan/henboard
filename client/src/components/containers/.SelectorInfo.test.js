import React from "react";
import SelectorInfo from "./SelectorInfo";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import { store } from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("SelectorInfo renders correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <SelectorInfo project="henboard" title="Todo" />
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
