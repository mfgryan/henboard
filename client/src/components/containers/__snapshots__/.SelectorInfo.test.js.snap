exports[`test SelectorInfo renders correctly 1`] = `
<div
  className="dropdown btn-group">
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-default"
    disabled={false}
    id="Selector"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    type="button">
    Send To
  </button>
  <ul
    aria-labelledby="Selector"
    className="dropdown-menu"
    role="menu">
    <li
      className=""
      role="presentation"
      style={undefined}>
      <a
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex="-1">
        1
          
        <small>
          05/15/17
        </small>
      </a>
    </li>
    <li
      onKeyDown={[Function]}
      onSelect={[Function]}>
      <a
        href="#"
        onClick={[Function]}>
        <b>
          load all
        </b>
      </a>
    </li>
    <li
      className="divider"
      onKeyDown={[Function]}
      role="separator"
      style={undefined} />
  </ul>
</div>
`;
